Assignment 3.Rmd

```diff
@@ -127,27 +129,94 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
 
 ```{r}
-
-
+g<-ggplot(data=K6,aes(week,avg))
+g+geom_line(aes(color=cluster))+xlab('Week')+ylab('Average Motivation')
 ```
 
 What patterns do you see in the plot?
 
+the average of motivation is higher for cluster 2 than 1
+
+
 It would be useful to determine how many people are in each cluster. We can do this easily with dplyr.
 
 ```{r}
-K7 <- 
+count(K5,cluster)
 ```
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
+```{r}
+fit1 <- kmeans(K3,3)
+fit1$cluster
+K4<-data.frame(K3,fit1$cluster)
+names(K4)<-c('1','2','3','4','5',"cluster")
+
+K5 <- tidyr::gather(K4,key="week",value='motivation',c(1:5))
+K6 <- dplyr::group_by(K5,week,cluster)%>%summarise(avg=mean(motivation))
+K6$week <- as.numeric(K6$week)
+
+K6$cluster <- as.factor  (K6$cluster)
+g<-ggplot(data=K6,aes(week,avg))
+g+geom_line(aes(color=cluster))+xlab('Week')+ylab('Average Motivation')
+
+#two cluster is better than 3
+```
 
 Once you have done this, save both of your plots to the Assignment 5 file. Create a Zotero item to index your new computer program (Assignment 5.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
 
 ##Part II
 
 Now, try to do the same for the data collected in class. Create two sets of clusters, the answers to the questions and regions where people grew up. 
+```{r}
+
+course<-read.csv("HUDK405018-cluster.csv")
+course<-dplyr::select(course,-c(1:17))
+course<-dplyr::slice(course,-c(1,2))
+for (i in 2:length(colnames(course))){
+  course[,i]=  as.numeric(course[,i])
+}
+course=na.omit(course)
+# find the latitude and longitude
+library(ggmap)
+region <- geocode(as.character(course$Q1_1), output = "latlon", source = "dsk")
+fit_region<-kmeans(region,2)
+questions<-course[,3:9]
+fit_questions<-kmeans(questions,2)
+fit_region$cluster
+fit_questions$cluster
+Q<-data.frame(questions,clusterQ=fit_questions$cluster)
+R<-data.frame(region,clusterR=fit_region$cluster)
+library(dplyr)
+
+colnames(Q)=c(1,2,3,4,5,6,7,"ClusterQ")
+Q1 <- tidyr::gather(Q,key="question",value='time',c(1:7))
+Q2<-dplyr::group_by(Q1,question,ClusterQ)%>%summarise(avg=mean(time))
+Q2$question<-as.numeric(Q2$question)
+
+Q2$ClusterQ=as.factor(Q2$ClusterQ)
+
+ggplot(data=Q2,aes(question,avg))+geom_line(aes(color=ClusterQ))
+
+R$lon=as.numeric(R $lon)
+R$lat=as.numeric(R$lat)
+R$clusterR=as.factor(R$clusterR)
+ggplot(data=R,aes(lon,lat))+geom_point(aes(color=clusterR))
+
+
+```
+
+
 
 ##Part III
 
 Create a visualization that shows the overlap between the two groups of clusters you created in part III.
+```{r}
+
+Total<-data.frame(Q,R)
+Total$ClusterQ=as.factor(Total$ClusterQ)
+Total$clusterR=as.factor(Total$clusterR)
+
+ggplot(data=Total,aes(lat,lon))+geom_point(aes(color=clusterR,shape=ClusterQ))
+
```

Comment:
Can you interpret what you have generated here? Is there overlap between the two groups of clusters?